File: src/main/osu/beatmap/constants/BeatmapCountdown.java
package main.osu.beatmap.constants;

/**
 * Represents the speed of the countdown before the first hit object.
 */
public enum BeatmapCountdown {
    noCountdown(0),
    normal(1),
    half(0.5f),
    twice(2);

    /**
     * The speed at which the beatmap countdown should be played.
     */
    public final float speed;

    /**
     * @param speed The speed at which the beatmap countdown should be played.
     */
    BeatmapCountdown(float speed) {
        this.speed = speed;
    }

    /**
     * Converts a string data from a beatmap file to its enum counterpart.
     *
     * @param data The data to convert.
     * @return The enum representing the data.
     */
    public static BeatmapCountdown parse(final String data) {
        switch (data) {
            case "0":
                return noCountdown;
            case "2":
                return half;
            case "3":
                return twice;
            default:
                return normal;
        }
    }
}

File: src/com/rian/osu/beatmap/constants/BeatmapCountdown.kt
package com.rian.osu.beatmap.constants

/**
 * Represents the speed of the countdown before the first hit object.
 */
enum class BeatmapCountdown(
    /**
     * The speed at which the beatmap countdown should be played.
     */
    @JvmField
    val speed: Float
) {
    NoCountdown(0f),
    Normal(1f),
    Half(0.5f),
    Twice(2f);

    companion object {
        /**
         * Converts a string data from a beatmap file to its enum counterpart.
         *
         * @param data The data to convert.
         * @return The enum representing the data.
         */
        @JvmStatic
        fun parse(data: String?) =
            when (data) {
                "0" -> NoCountdown
                "2" -> Half
                "3" -> Twice
                else -> Normal
            }
        }
}
